### PR TITLE
Add option to support namespacetopic

### PR DESCRIPTION
**Pull request by DArdouin**
## Description of changes
Added support for Topic name and topic type un publisher options

## Breaking changes
None

## Additional checks
- N/A Updated the documentation of the project to reflect the changes
- [x] Added new tests that cover the code changes

### CODE REVIEW

File: src/Workleap.DomainEventPropagation.Publishing.Tests/EventPropagationPublisherOptionsValidatorTests.cs
Comment (felpel) [inline src/Workleap.DomainEventPropagation.Publishing.Tests/EventPropagationPublisherOptionsValidatorTests.cs:1]: You should test scenarios where a failure should be raised (e.g. Have a topic type set to "Namespace" and the topic name would be null or empty)

File: src/Workleap.DomainEventPropagation.Publishing/EventPropagationPublisherOptions.cs [hunk 1]
```diff
@@ -13,4 +19,8 @@ public sealed class EventPropagationPublisherOptions
     public TokenCredential? TokenCredential { get; set; }
 
     public string TopicEndpoint { get; set; } = string.Empty;
+
+    public string TopicName { get; set; } = string.Empty;
```
Comment (felpel) [inline src/Workleap.DomainEventPropagation.Publishing/EventPropagationPublisherOptions.cs:23]: `TopicName` should be of type `string?`, considering it _can_ be `null` when `TopicType` is set to `TopicType.Default`

```suggestion
    public string? TopicName { get; set; }
```

File: src/Workleap.DomainEventPropagation.Publishing/PublicAPI.Shipped.txt
Comment (felpel) [inline src/Workleap.DomainEventPropagation.Publishing/PublicAPI.Shipped.txt:1]: Similar comment that I left in #72

**Review by felpel — CHANGES_REQUESTED** [review]

File: src/Workleap.DomainEventPropagation.Publishing/EventPropagationPublisherOptions.cs (hunk 1 above)
Comment (DArdouin) [reply src/Workleap.DomainEventPropagation.Publishing/EventPropagationPublisherOptions.cs:23]: Good point

**Review by felpel — APPROVED** [review]

**Pull request merged**